trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,350 @@
+            allocate ( this%output )
+            if ( .not. associated(this%output) ) then
+                ! if the output structure could not be created write an error
+                write(0,*) "Failure to allocate level pool output structure"
+            else
+                ! initialize the output structure
+                call this%output%init()
+            end if
+
+            ! try to allocate parameters
+            allocate ( this%parameters )
+            if ( .not. associated(this%parameters) ) then
+                ! if the parameters structure could not be created write an error
+                write(0,*) "Failure to allocate hybrid parameters structure"
+            else
+                ! initialize hybrid parameters
+                call this%parameters%init(lake_area, lake_max_water_elevation, orifice_elevation, lake_number, &
+                observation_lookback_hours, observation_update_time_interval_seconds, persistence_parameter_file)
+            end if
+            this%pointer_allocation_guard = .true.
+
+            ! try to allocate state
+            allocate ( this%state )
+            if ( .not. associated(this%state) ) then
+                ! if the state structure could not be created write an error
+                write(0,*) "Failure to allocate state parameters structure"
+            else
+                ! initialize hybrid state
+                call this%state%init(water_elevation, lake_area, lake_max_water_elevation, orifice_elevation, initial_fractional_depth)
+            end if
+            this%pointer_allocation_guard = .true.
+
+            ! Initialize persistence weight index to be out of bounds of the persistence weights array
+            this%state%persistence_weight_index = SIZE(this%parameters%persistence_weighted_coefficients) + 1
+
+            ! Call hydro stop if the initial storage is greater than max storage or less than zero.
+            if (this%state%current_storage > this%parameters%max_storage) then
+                write(lake_number_string, "(I15)") lake_number
+                call hydro_stop("Initial storage exceeds max storage for reservoir " // lake_number_string)
+
+            else if (this%state%current_storage < this%parameters%min_storage) then
+                write(lake_number_string, "(I15)") lake_number
+                call hydro_stop("Initial storage is less than zero for reservoir " // lake_number_string)
+
+            end if
+
+            ! Allocate a single level pool reservoir
+            allocate(levelpool_struct :: this%state%levelpool_ptr)
+
+            ! Initialize level pool reservoir
+            call this%state%levelpool_ptr%init(water_elevation, lake_area, &
+            weir_elevation, weir_coeffecient, weir_length, orifice_elevation, &
+            orifice_coefficient, orifice_area, lake_max_water_elevation, lake_number)
+
+        end if
+
+    end subroutine hybrid_struct_init
+
+
+    !Level Pool Destructor
```

Comment:
i'd remove the comment... seems stale from a copy

Reply:
@jdmattern-noaa fixing. Changing to 'Hybrid'.